src/allmydata/client.py

```diff
@@ -533,35 +530,35 @@ def init_web(self, webport):
         self.log("init_web(webport=%s)", args=(webport,))
 
         from allmydata.webish import WebishServer
-        nodeurl_path = os.path.join(self.basedir, "node.url")
-        staticdir_config = self.get_config("node", "web.static", "public_html").decode("utf-8")
-        staticdir = abspath_expanduser_unicode(staticdir_config, base=self.basedir)
+        nodeurl_path = self.config.get_config_path("node.url")
+        staticdir_config = self.config.get_config("node", "web.static", "public_html").decode("utf-8")
+        staticdir = self.config.get_config_path(staticdir_config)
```

Comment:
I like the look of this `staticdir ...` code transformation but is it definitely equivalent?  `get_config_path` does much less than `abspath_expanduser_unicode`.  It _seems_ like it should be sufficient but I haven't messed around with paths in tahoe.cfg much.  I don't have much in the way of expectations about how these values get interpreted (particularly not when it comes to weird unicode paths and Windows-specific stuff).

Reply:
Hmm, yes, `_Config` now does `abspath_expanduser_unicode` on its `_basedir`, so `get_config_path` should be an expanded unicode path ... but in this case we're taking user input, so they could do something arbitrarily crazy like configuring "../../../../../foo" or whatever. So, it should probably be expanded again. Nice catch :)